File: typescript-basic/classes/adapter1.ts
// Klasa Adaptee (Stare API)
class OldAPI {
    getData(): { firstname: string; lastname: string; age: string; id: string } {
        return {
            firstname: "John",
            lastname: "Doe",
            age: "30", // Wiek jako string
            id: "12345", // ID jako string
        };
    }
}

// Target Interface (Nowe API)
interface NewAPI {
    getFormattedData(): { fullname: string; age: number; id: number };
}

// Adapter
class OldToNewAPIAdapter implements NewAPI {
    private oldAPI: OldAPI;

    constructor(oldAPI: OldAPI) {
        this.oldAPI = oldAPI;
    }

    getFormattedData(): { fullname: string; age: number; id: number } {
        const oldData = this.oldAPI.getData();

        // Konwersja danych do formatu wymaganego przez Nowe API
        return {
            fullname: `${oldData.firstname} ${oldData.lastname}`, // Łączenie firstname i lastname
            age: parseInt(oldData.age, 10), // Konwersja wieku na liczbę
            id: parseInt(oldData.id, 10),  // Konwersja ID na liczbę
        };
    }
}

// Klient (Nowa funkcjonalność)
function useNewAPI(api: NewAPI) {
    const data = api.getFormattedData();
    console.log("Sformatowane dane:", data);
}

// Przykład użycia
const oldAPI = new OldAPI();
const adapter = new OldToNewAPIAdapter(oldAPI);

useNewAPI(adapter);

File: python/102/patterns/adapter1.py
class OldAPI:
    def get_data(self):
        return {
            "firstname": "John",
            "lastname": "Doe",
            "age": "30",  # Wiek jako string
            "id": "12345"  # ID jako string
        }

# Adapter
class OldToNewAPIAdapter:
    def __init__(self, old_api):
        self.old_api = old_api

    def get_formatted_data(self):
        old_data = self.old_api.get_data()
        return {
            "fullname": f"{old_data['firstname']} {old_data['lastname']}",
            "age": int(old_data["age"]),  # Konwersja wieku na liczbę
            "id": int(old_data["id"])    # Konwersja ID na liczbę
        }

# Klient
def use_new_api(api):
    data = api.get_formatted_data()
    print("Sformatowane dane:", data)

# Przykład użycia
old_api = OldAPI()
adapter = OldToNewAPIAdapter(old_api)

use_new_api(adapter)